flink-table-store-connector/src/test/java/org/apache/flink/table/store/connector/ReadWriteTableITCase.java

```diff
@@ -1713,4 +1748,74 @@ private static StreamExecutionEnvironment buildBatchEnv() {
         env.setParallelism(2);
         return env;
     }
+
+    @Test
+    public void testSourceParallelism() throws Exception {
+        String managedTable = createSourceAndManagedTable(false, false, false, false, false).f1;
+
+        // without hint
+        String query = prepareSimpleSelectQuery(managedTable, Collections.emptyMap());
+        assertThat(sourceParallelism(query)).isEqualTo(env.getParallelism());
+
+        // with hint
+        query =
+                prepareSimpleSelectQuery(
+                        managedTable, Collections.singletonMap(SCAN_PARALLELISM.key(), "66"));
+        System.out.println(query);
+        assertThat(sourceParallelism(query)).isEqualTo(66);
+    }
+
+    private int sourceParallelism(String sql) {
+        DataStream<Row> stream = tEnv.toChangelogStream(tEnv.sqlQuery(sql));
+        return stream.getParallelism();
+    }
+
+    @Test
+    public void testSinkParallelism() {
+        testSinkParallelism(null, env.getParallelism());
+        testSinkParallelism(23, 23);
+    }
+
```

Comment:
Nit: group test methods and util methods together?

Reply:
this is for testing both default and hint